ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PingPacketData.java

```diff
@@ -56,48 +59,62 @@ private PingPacketData(
     this.enrSeq = enrSeq;
   }
 
-  public static PingPacketData create(final Endpoint from, final Endpoint to, final UInt64 enrSeq) {
+  public static PingPacketData create(
+      final Optional<Endpoint> from, final Endpoint to, final UInt64 enrSeq) {
     checkArgument(
         enrSeq != null && UInt64.ZERO.compareTo(enrSeq) < 0, "enrSeq cannot be null or negative");
     return create(from, to, PacketData.defaultExpiration(), enrSeq);
   }
 
   static PingPacketData create(
-      final Endpoint from, final Endpoint to, final long expirationSec, final UInt64 enrSeq) {
+      final Optional<Endpoint> from,
+      final Endpoint to,
+      final long expirationSec,
+      final UInt64 enrSeq) {
     checkArgument(
         enrSeq != null && UInt64.ZERO.compareTo(enrSeq) < 0, "enrSeq cannot be null or negative");
-    return new PingPacketData(Optional.of(from), to, expirationSec, enrSeq);
+    return new PingPacketData(from, to, expirationSec, enrSeq);
   }
 
   public static PingPacketData readFrom(final RLPInput in) {
     in.enterList();
     // The first element signifies the "version", but this value is ignored as of EIP-8
     in.readBigIntegerScalar();
-    final Optional<Endpoint> from = Endpoint.maybeDecodeStandalone(in);
-    final Endpoint to = Endpoint.decodeStandalone(in);
+    Optional<Endpoint> from = Optional.empty();
+    Optional<Endpoint> to = Optional.empty();
+    if (in.nextIsList()) {
+      to = Endpoint.maybeDecodeStandalone(in);
```

Comment:
are you managing the case where the list is present but we don't have an IP, just the port (from field)

Reply:
nope, i did not consider that. I'll have to add that.